sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/SpringServiceImpl.java

```diff
@@ -107,70 +121,80 @@ public SpringServiceImpl withSku(String skuName, int capacity) {
 
     @Override
     public SpringServiceImpl withSku(Sku sku) {
+        needUpdate = true;
         inner().withSku(sku);
         return this;
     }
 
     @Override
     public SpringServiceImpl withTracing(String appInsightInstrumentationKey) {
-        if (inner().properties() == null) {
-            inner().withProperties(new ClusterResourceProperties());
-        }
-        inner().properties().withTrace(
-            new TraceProperties().withAppInsightInstrumentationKey(appInsightInstrumentationKey).withEnabled(true));
+        this.addPostRunDependent(
+            context -> manager().inner().getMonitoringSettings()
+                .updatePatchAsync(resourceGroupName(), name(), new MonitoringSettingProperties()
+                    .withAppInsightsInstrumentationKey(appInsightInstrumentationKey)
+                    .withTraceEnabled(true))
+                .then(context.voidMono())
+        );
```

Comment:
Better to have an object configured, and delay postRunDependency to e.g. `beforeCreate`